File: openapi-test-kotlin-server-jaxrs/testapi/model/Dog.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import java.time.OffsetDateTime
import java.util.Optional

data class Dog(
  @JsonProperty("id") @NotNull val id: String,
  @JsonProperty("name") @NotNull val name: String,
  @JsonProperty("updatedAt") val updatedAt: Optional<OffsetDateTime>,
  @JsonProperty("breed") @NotNull val breed: String,
  @JsonProperty("createdAt") @NotNull val createdAt: OffsetDateTime,
  @JsonProperty("barkVolume") val barkVolume: Optional<Integer>
) : Animal {
  override fun animal_type(): String = "dog"
}

File: openapi-test-scala-server-spring-client-jdk/testapi/model/Dog.scala
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.Optional

case class Dog(
  @JsonProperty("id") id: PetId,
  @JsonProperty("name") name: String,
  @JsonProperty("updatedAt") updatedAt: Optional[OffsetDateTime],
  @JsonProperty("breed") breed: String,
  @JsonProperty("createdAt") createdAt: OffsetDateTime,
  @JsonProperty("barkVolume") barkVolume: Optional[Integer]
) extends Animal {
  @JsonProperty("animal_type")
  override lazy val animal_type: String = "dog"
}